proposals/Fleetctl-Docker-Image.md

```diff
@@ -0,0 +1,51 @@
+# Goal
+
+We need `fleetctl package` functionality to generate all types of packages (PKG, MSI, DEB and RPM) from Linux.
+
+# How
+
+Create a new Docker image `fleetdm/fleetctl` that will contain `fleetctl` and all the dependencies ready to create packages.
```

Comment:
@lucasmrod a few thoughts/questions as I'm building this:

- Feels like the `fleetctl` binary that will be included in the Docker image should be a released version that we `curl`  as opposed to running a `COPY` command to include a local build. Do you have any thoughts? this feels safer, but comes with two drawbacks:
    - a dependency in the release process
    - for local development, we will need to run a local file server or edit the Dockerfile temporarily
- Do we want to deprecate `fleetdm/wix` and `fleetdm/bomutils`? feels like we could replace all the code that uses both images to use `fleetdm/fleetctl` instead. If the answer is 'yes', is that part of the scope?

Reply:
> Feels like the fleetctl binary that will be included in the Docker image should be a released version that we curl as opposed to running a COPY command to include a local build. Do you have any thoughts? this feels safer, but comes with two drawbacks

Why not use the `COPY` approach? (Maybe am missing something.)

> Do we want to deprecate `fleetdm/wix` and `fleetdm/bomutils`? feels like we could replace all the code that uses both images to use fleetdm/fleetctl instead. If the answer is 'yes', is that part of the scope?

New versions of `fleetctl` should not use `fleetdm/{wix|bomutils}`, but we should not deprecate them because old/current versions of `fleetctl` will still rely on them.

Reply:
Also, I believe `--native` should only work when `GOOS=linux`.

Reply:
> Why not use the `COPY` approach? (Maybe am missing something.)

both are valid, just that with `curl` we are completely sure that we use the exact same binary that was released. This is probably less of a concern if we have CI build and release the docker images for us.

> New versions of fleetctl should not use `fleetdm/{wix|bomutils}`

Is this in the scope of https://github.com/fleetdm/fleet/issues/6363 and https://github.com/fleetdm/fleet/issues/6364? is there a separate ticket? I'm making sure I understand what needs to be done in this iteration.

> we should not deprecate them because old/current versions of fleetctl will still rely on them.

If the new versions of `fleetctl` from now on don't use these images at all, why not deprecate? Just to overcommunicate: by deprecate I don't mean remove them from the registry, just communicate that they shouldn't be actively maintained anymore.

Reply:
> both are valid, just that with curl we are completely sure that we use the exact same binary that was released. This is probably less of a concern if we have CI build and release the docker images for us.

Correct. We will eventually release the `fleetdm/fleetctl` image. Initially will probably be used internally by us.

> Is this in the scope of https://github.com/fleetdm/fleet/issues/6363 and https://github.com/fleetdm/fleet/issues/6364? is there a separate ticket? I'm making sure I understand what needs to be done in this iteration.

Correct. New versions of `fleetctl` will run with native tooling (for WiX and Bom) iff `GOOS=linux && FLEETCTL_NATIVE=1`.

> If the new versions of fleetctl from now on don't use these images at all, why not deprecate? Just to overcommunicate: by deprecate I don't mean remove them from the registry, just communicate that they shouldn't be actively maintained anymore.

Yes, we are on the same page there (not deleted and just not maintained anymore).

Reply:
❤️ thank you!